README.md

```diff
@@ -1,34 +1,194 @@
-[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_template_plugin)
-[![CircleCI](https://circleci.com/gh/membraneframework/membrane_template_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_template_plugin)
+[![Hex.pm](https://img.shields.io/hexpm/v/membrane_telemetry_metrics.svg)](https://hex.pm/packages/membrane_telemetry_metrics)
+[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_telemetry_metrics)
+[![CircleCI](https://circleci.com/gh/membraneframework/membrane_telemetry_metrics.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_telemetry_metrics)
 
-This repository contains a template for new plugins.
-
-Check out different branches for other flavours of template.
+This repository contains a tool for generating metrics.
 
 It is part of [Membrane Multimedia Framework](https://membraneframework.org).
 
 ## Installation
 
-The package can be installed by adding `membrane_template_plugin` to your list of dependencies in `mix.exs`:
+The package can be installed by adding `membrane_telemetry_metrics` to your list of dependencies in `mix.exs`:
 
 ```elixir
 def deps do
   [
-    {:membrane_template_plugin, "~> 0.1.0"}
+    {:membrane_telemetry_metrics, github: "membraneframework/membrane_telemetry_metrics"}
   ]
 end
 ```
 
 ## Usage
 
-TODO
+To benefit from of full functionality offered by this tool, you have to use both `Membrane.TelemetryMetrics` and `Mebrane.TelemetryMetrics.Reporter`. 
+
+First, put 
+```elixir
+config :membrane_telemetry_metrics, emit_events: :all
+```
+in your `config.exs`. Alternatively, instead of `:all`, you can also pass there a list of emitted events, in this case, it will be `[[:shopping]]`
+
+
+Let's assume, that you want to track three metrics: 
+ * `cash_spent`
+ * `number_of_payments`
+ * `last_visited_shop`
+So, the definition of our list of metrics will look like 
+```elixir 
+metrics = 
+  [
+    Telemetry.Metrics.sum(
+      "cash_spent",
+      event_name: [:shopping],
+      measurement: :payment
+    ),
+    Telemetry.Metrics.counter(
+      "number_of_payments",
+      event_name: [:shopping]
+    ),
+    Telemetry.Metrics.last_value(
+      "last_visited_shop",
+      event_name: [:shopping],
+      measurement: :shop
+    )
+  ]
+```
+
+Then, you have to start `Mebrane.TelemetryMetrics.Reporter`. It can be made by calling 
+```elixir
+  Membrane.TelemetryMetrics.Reporter.start_link([metrics: metrics], name: ShoppingReporter)
+```
+but the suggested way is to do it under `Supervisor` tree.
+
+Now, `ShoppingReporter` is ready to aggregate metrics from emitted events. But before that, we have to register our event in every process, that will emit it. Let's assume, that we want to aggregate data about three people: `James Smith`, `Mary Smith` and `Patricia Johnson`. Events with data about each of these people will be emitted in a different process.
```

Comment:
I would put some introduction at the beginning of this example to explain what we want to achieve and then how we can achieve this using `Membrane.TelemetryMetrics`